pageserver/src/tenant/timeline.rs

```diff
@@ -3334,6 +3386,12 @@ impl Timeline {
 
         let mut cont_lsn = Lsn(request_lsn.0 + 1);
 
+        let mut read_path = if cfg!(feature = "testing") || cfg!(debug_assertions) {
+            Some(ReadPath::new(keyspace.clone(), request_lsn))
+        } else {
+            None
+        };
```

Comment:
Wdyt of gating this behind a config flag instead? It might be useful to run this with production builds on a testing box.

Reply:
https://github.com/neondatabase/neon/pull/10528/commits/171663ccab7534fcddcfa2c496a7a3166bee0c54